crates/sui-types/src/gas.rs

```diff
@@ -350,6 +350,9 @@ pub struct SuiGasStatus<'a> {
     /// was the storage cost paid when the object was last mutated. It is not affected
     /// by the current storage gas unit price.
     storage_rebate: SuiGas,
+    /// Amount of storage rebate accumulated when we are running in unmetered mode (i.e. system transaction).
+    /// This allows us to track how much storage rebate we need to retain in system transactions.
+    unmetered_storage_rebate: SuiGas,
```

Comment:
Tracking in a separate field is easier to deal with. But it should certainly be possible to track it in `storage_rebate`.
For example, we want to make sure that the resulting gas summary still have storage_rebate=0, which is important because nobody is getting these rebates (they all go to the next object).